### PR TITLE
Improve thrift encoder 's encode_limited_size

### PR DESCRIPTION
**Pull request by luong-komorebi**
This improve https://github.com/salemove/jaeger-client-ruby/pull/22. I was originally planned to continue that PR but later found that it is merged into master via another commit.

### CODE REVIEW

File: spec/jaeger/encoders/thrift_encoder_spec.rb [hunk 1]
```diff
@@ -1,5 +1,20 @@
 require 'spec_helper'
 
+RSpec::Matchers.define :be_a_valid_span do |_|
```
Comment (indrekj) [inline spec/jaeger/encoders/thrift_encoder_spec.rb]: Lets use here `be_a_valid_thrift_span`

File: spec/jaeger/encoders/thrift_encoder_spec.rb [hunk 2]
```diff
@@ -46,4 +61,45 @@
...
+      Jaeger::SpanContext.new(
+        trace_id: Jaeger::TraceId.generate,
+        span_id: Jaeger::TraceId.generate,
+        flags: Jaeger::SpanContext::Flags::DEBUG
+      )
+    end
+    let(:example_span) { Jaeger::Span.new(context, 'example_op', nil) }
+
+    it 'encode successfully' do
+      encoded_span = encoder.encode([example_span])
```
Comment (indrekj) [inline spec/jaeger/encoders/thrift_encoder_spec.rb]: I think this should be `encoded_batch =`

File: spec/jaeger/encoders/thrift_encoder_spec.rb [hunk 3]
```diff
@@ -46,4 +61,45 @@
...
+    let(:context) do
+      Jaeger::SpanContext.new(
+        trace_id: Jaeger::TraceId.generate,
+        span_id: Jaeger::TraceId.generate,
+        flags: Jaeger::SpanContext::Flags::DEBUG
+      )
+    end
+    let(:example_span) { Jaeger::Span.new(context, 'example_op', nil) }
+
+    it 'encode successfully' do
```
Comment (indrekj) [inline spec/jaeger/encoders/thrift_encoder_spec.rb]: `it 'encodes spans into one batch'`

File: spec/jaeger/encoders/thrift_encoder_spec.rb [hunk 4]
```diff
@@ -46,4 +61,45 @@
...
+        span_id: Jaeger::TraceId.generate,
+        flags: Jaeger::SpanContext::Flags::DEBUG
+      )
+    end
+    let(:example_span) { Jaeger::Span.new(context, 'example_op', nil) }
+    # example span have size of 63, so let's make an array of 100 span
+    # and set the limit at 5000, so it should return a batch of 2
+    let(:example_spans) { Array.new(100, example_span) }
+
+    it 'encode successfully in batches' do
```
Comment (indrekj) [inline spec/jaeger/encoders/thrift_encoder_spec.rb]: `it 'encodes spans into multiple batches'`

**Comment by luong-komorebi** [comment]
Hi @indrekj , thank you for your review. I addressed all of them in the latest commit.

File: spec/jaeger/encoders/thrift_encoder_spec.rb [hunk 5]
```diff
@@ -91,15 +91,14 @@
...
-      expect(encoded_span_batches.first.spans.first).to be_a_valid_span
-      expect(encoded_span_batches.first.spans.first).to be_a_valid_span
-      expect(encoded_span_batches.first.spans.last).to be_a_valid_span
-      expect(encoded_span_batches.last.spans.first).to be_a_valid_span
-      expect(encoded_span_batches.last.spans.last).to be_a_valid_span
+    it 'encodes spans into multiple batches' do
+      encoded_batches = encoder.encode_limited_size(example_spans, ::Thrift::CompactProtocol, 5_000)
+      expect(encoded_batches.length).to be(2)
+      expect(encoded_batches.first.spans.first).to be_a_valid_thrift_span
+      expect(encoded_batches.first.spans.first).to be_a_valid_thrift_span
```
Comment (indrekj) [inline spec/jaeger/encoders/thrift_encoder_spec.rb]: duplicate line

**Comment by indrekj** [comment]
Merged,

I squashed last 3 commits.

These are in master now:
346c23f14c6b3bb88bb9884bcaed01332e31c35d
16c31d6636f5712a4d12ba54d9cd3320d5c60048
6f4923d7c6a8a932c5ac7fdb99fba67b9a3fe0fc

Thank you.

**Pull request closed without merging**